src/middlewared/middlewared/plugins/system/vendor/vendor.py

```diff
@@ -0,0 +1,27 @@
+import os
+
+from middlewared.api import api_method
+from middlewared.api.current import NoArgs, NoneReturn, VendorNameResult
+from middlewared.service import Service
+
+
+SENTINEL_FILE_PATH = '/data/.vendor'
+
+
+class VendorService(Service):
+
+    @api_method(NoArgs, VendorNameResult, private=True)
+    def name(self) -> str | None:
+        try:
+            with open(SENTINEL_FILE_PATH, 'r') as file:
+                if contents := file.read():
+                    return contents
+        except FileNotFoundError:
+            return None
+
+    @api_method(NoArgs, NoneReturn, private=True)
```

Comment:
We can’t use this approach. If we decide to add arguments or return value later, changing model names here will have a retroactive effect on already released API. Unfortunately, we’ll have to be very repetitive and declare empty arguments and empty result for each method that accepts/returns them separately.

Reply:
That is unfortunate but I get your point.